### PR TITLE
Atom feed

### PR DESCRIPTION
**Pull request by stephencelis**
Preliminary work for Atom/RSS feeds.

### CODE REVIEW

**Comment by stephencelis** [comment]
(Still a WIP, but getting there.)

File: Sources/PointFree/AtomFeed.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+let pointFreeFeed = View<[Episode]> { episodes in
+  atomLayout.view(
+    AtomFeed(
+      author: AtomAuthor(
+        email: "support@pointfree.co",
+        name: "Point-Free"
+      ),
+      entries: episodes.map(atomEntry(for:)),
+      atomUrl: url(to: .feed(.atom)),
+      siteUrl: "/", // FIXME url(to: .root),
```
Comment (mbrandonw) [inline Sources/PointFree/AtomFeed.swift]: can't we make this `.secretHome` or `.home(signedUpSuccessfully: nil)` right now?

File: Sources/PointFree/AtomFeed.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import Foundation
+import Html
+import HttpPipeline
+import MediaType
+import Prelude
+
+let atomFeedResponse =
+  writeStatus(.ok)
+    >-> respond(pointFreeFeed, contentType: .html)
```
Comment (mbrandonw) [inline Sources/PointFree/AtomFeed.swift]: is the content type supposed to be xml?

File: Sources/PointFree/AtomFeed.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    conn |> respond(body: view.rendered(with: conn.data), contentType: contentType)
+  }
+}
+
+private func atomEntry(for episode: Episode) -> AtomEntry {
+  return AtomEntry(
+    title: episode.title,
+    siteUrl: url(to: .episode(.left(episode.slug))),
+    updated: episode.publishedAt,
+    content: []
```
Comment (mbrandonw) [inline Sources/PointFree/AtomFeed.swift]: should we use the `episode.blurb` here? or we might even be able to reuse the exact view from the "new episode" email. are atom entries allowed to have inline css styles?

File: Sources/PointFree/Atom.swift [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+public func content(_ attribs: [Attribute<Element.Content>], _ content: [Node]) -> Node {
+  return node("content", attribs, [.text(unsafeUnencodedString("<![CDATA[" + render(content).string + "]]>"))])
+}
+
+public func type(_ type: String) -> Attribute<Element.Content> {
+  return attribute("type", type)
+}
+
+private let atomDateFormatter = DateFormatter() // FIXME
```
Comment (mbrandonw) [inline Sources/PointFree/Atom.swift]: what needs to be fixed here?

File: Sources/PointFree/AtomFeed.swift (hunk 2 above)
Comment (stephencelis) [reply Sources/PointFree/AtomFeed.swift]: It is! Or `application/atom+xml`/`application(.atom(.xml))` Needs to be fixed.

File: Sources/PointFree/AtomFeed.swift (hunk 1 above)
Comment (stephencelis) [reply Sources/PointFree/AtomFeed.swift]: Yep yep.

File: Sources/PointFree/Atom.swift (hunk 4 above)
Comment (stephencelis) [reply Sources/PointFree/Atom.swift]: Needs to be configured for Atom's formatting.

File: Sources/PointFree/AtomFeed.swift (hunk 3 above)
Comment (stephencelis) [reply Sources/PointFree/AtomFeed.swift]: I don't think they'd render. Atom entries should just be basic DOM HTML.

File: Sources/PointFree/AtomFeed.swift (hunk 2 above)
Comment (mbrandonw) [reply Sources/PointFree/AtomFeed.swift]: `MediaType` should prob be switched to a a struct huh? or at least have an `other` case?

**Pull request merged**